waiter/src/waiter/service_description.clj

```diff
@@ -48,6 +48,7 @@
    (s/required-key "version") schema/non-empty-string
    (s/required-key "run-as-user") schema/non-empty-string
    ;; Optional
+   (s/optional-key "allowed-params") #{schema/non-empty-string}
```

Comment:
Could the schema enforce comma separated list, and restrict the param names [A-Za-z\-_]+?